source/resource_name.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/common/resource_name.h>
+
+#include <aws/common/common.h>
+#include <aws/common/byte_buf.h>
+
+#define ARN_DELIMETER ":"
+
+static int s_parse_resource_name_part(struct aws_byte_cursor *token_cursor, const struct aws_byte_cursor *cursor_split_on, struct aws_byte_cursor *prev_token_cursor) {
+    prev_token_cursor->ptr = token_cursor->ptr + 1;
+    prev_token_cursor->len = token_cursor->len;
```

Comment:
Shouldn't this be len - 1?  Alternatively you could assign it with token_cursor and advance 1